### PR TITLE
Fix for issue #1237

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by hanw** [comment]
@mbudiu-vmw Do you want to rebase this one?

File: testdata/p4_14_samples_outputs/issue1237-first.p4 [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        hdr.kilometer.sleeting = hdr.kilometer.sleeting + 16w65529;
+    }
+    @name(".conceptualization") table conceptualization {
+        actions = {
+            mooneys();
+            @defaultonly NoAction();
+        }
+        key = {
+            hdr.kilometer.isValid()      : exact @name("kilometer.$valid$") ;
+            48w0                         : lpm @name("0") ;
```
Comment (hanw) [inline testdata/p4_14_samples_outputs/issue1237-first.p4]: It's unclear to me if the control plane API for this argument would be generated correctly. Would the argument be named '0'?

Comment (mbudiu-vmw) [reply testdata/p4_14_samples_outputs/issue1237-first.p4]: That's a good question. Today this is a warning, maybe it should be an error?
@antoninbas : what should happen when a key field is a constant expression?

Comment (jafingerhut) [reply testdata/p4_14_samples_outputs/issue1237-first.p4]: I have a hard time imagining a situation where someone intentionally wants a compile-time constant value as a table key expression.  It seems that a correct implementation could leave that key in, and always search with that field equal to the constant value (0 in this example), but if the control plane ever installed a table entry that did not match 0 for that field, such a table entry would be impossible to match.

It seems like it justifies at least a warning from the compiler, if not an error.  If it is a warning, it isn't clear that much effort should be made in later tools to support this case, unless someone can think of a valid use case for it, so perhaps an error is preferable?

Comment (antoninbas) [reply testdata/p4_14_samples_outputs/issue1237-first.p4]: I think one acceptable solution that is probably also the easiest to implement is to print a warning in the p4RuntimeSerializer and omit the field altogether from the P4Info message. This way we do not put any burden on the P4Runtime implementers, for something that doesn't seem to have any actual use-cases.

Comment (mbudiu-vmw) [reply testdata/p4_14_samples_outputs/issue1237-first.p4]: Since we already have a warning I declare this PR ready for review.

**Pull request merged**